src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -484,6 +503,249 @@ protected void addRead(final int size) {
      * @param client This is the client the close is being called for.
      */
     public void onClose(Client client);
-  }  
+  }
+  
+  /**
+   * ClientOptions that can be changed depending on what kind of client you want.
+   * 
+   * In general these should not be set unless you have a very specific use case.
+   * 
+   * @author lwahlmeier
+   *
+   */
+  public interface ClientOptions {
+    
+    /**
+     * This is only available for connection backed by a TCP socket.
+     * It will turn on TcpNoDelay on the the connection at the System level.
+     * This means that data queued to go out the socket will not delay before being sent.
+     * 
+     * @param enabled true means NoDelay is on, false means NoDelay is off.
+     * @return true if this was able to be set.
```

Comment:
The whole returning true/false seems kind of awkward.  At first when reading the docs I assumed that a false should indicate that a retry for setting the value should be done (ie a CAS failed, and could not be set at this point, and thus should be retried at some other point).

I now realize that they are supposed to indicate if they _can_ be set.

I am not sure the best way to do this, but a few options:
* setting a value could be done by accepting an enum + value.  If you do that, you can then have a function "getAvailableOptions" which returns a set of possible accepted enum values.
* You could just ignore it if the class does not allow that value to be set.  Since you have to attempt setting it anyways to find out if it could be, I am not sure how an application could reasonably handle that information.
* You could add a third function for every value...."canSetX".

I do think any of the above options would make the API less confusing here.